www/guides/react.md

```diff
@@ -0,0 +1,350 @@
+  }
+  to {
+    transform: rotate(360deg);
+  }
+}
+```
+
+To use this CSS, head to `App.jsx` and import it
+
+```diff
+  import logo from './logo.svg';
++ import './App.css';
+```
+
+<div class="frame"><img src="/img/guides/react/react.gif" alt="The page now has centered items, a grey background, styled fonts, and the React logo has an animation that rotates it." class="screenshot"/></div>
+
+## Making Snowpack Even Faster with Fast Refresh
+
+[React Fast Refresh](https://reactnative.dev/docs/fast-refresh)? What's that? It's a developer tool that makes it much easier to see how updates to your code work in the browser. React projects are often interactive and include state. For example, this project you're building has a state that is the amount of time on the page. When developing with state it's useful not to lose it while you edit code. React Fast Refresh shows you updates without refreshing the entire page. Showing you how to add this is also a good intro to Snowpack plugins. Snowpack starts with a minimal setup with the perspective that you can add what you need through the plugin system.
+
+Start by enabling [Hot Module Replacement](https://www.snowpack.dev/#hot-module-replacement), a requirement for Fast Refresh. Snowpack automatically refreshes to show changes in your code. Hot Module Replacement only updates the components that changed without refreshing the whole page. You can enable it for React with a small snippet of code in `index.jsx`.
+
+Head to `index.jsx` and add this snippet at the bottom:
+
+```diff
+  ReactDOM.render(
+    <React.StrictMode>
+      <App />
+    </React.StrictMode>,
+    document.getElementById('root'),
+  );
++ // Hot Module Replacement (HMR) - Remove this snippet to remove HMR.
++ // Learn more: https://www.snowpack.dev/#hot-module-replacement
++ if (import.meta.hot) {
++   import.meta.hot.accept();
++ }
+```
+
+Now when you change `App.jsx` the changes show without the whole page refreshing.
+
+<div class="frame"><img src="/img/guides/react/hmr.gif" alt="GIF showing code side by side with the app. A change in made to App.jsx and it shows immediately when the file is changed. The counter keeps counting uninterrupted." class="screenshot"/></div>
+
+The counter on the page still resets to 0, even though the page itself hasn't fully refreshed. This can be annoying if you're trying to debug state. React Fast Refresh solves this.
+
+Installing a plugin involves installing the module and then enabling it in the Snowpack configuration file.
+
+Install the module first by running.
+
+```bash
+npm install @snowpack/plugin-react-refresh --save-dev
+```
+
+To enable it head to `snowpack-config.js` and add it to the array of plugins
+
+```diff
+  module.exports = {
+    mount: {
+      public: '/',
+      src: '/_dist_',
+    },
+-   plugins: []
++   plugins: ['@snowpack/plugin-react-refresh'],
+  };
+```
+
+> 💡 Tip: you'll need to restart Snowpack for the configuration changes to take effect
```

Comment:
instead of having this in a blockquote, start the next paragraph with this instruction. Blockquotes are only used above as additional info and not important/required steps.